Task-54252: Send funds using Metamask

MayTekayaa opened the pull request:
A connected user wants to send funds with metamask wallet either with meeds wallet, he first accesses the wallet application then clicks on the exchange button and then on sends funds button. If the user is connected with metamask, when he fills in the form with the recipient and the amount, a metamask popin will appear to confirm the transaction. and if the user is connected with the meeds wallet, the sending of transaction will be done by a simple click on the send button.

bourasom committed:
TASK-54163: Create FB meedsv2 and update projects versions/dependencies

ArfaouiMohamedAmine committed:
TASK-56288 : fix automatic scroll to wallet settings (#227)

No more scrolling down to wallet settings when opening user settings.
so added a window.location.hash parse to the extension registry wallet when it's mounted to add an anchor to the href and scroll down to the wallet whenever it's loaded .

GouadriaHanen committed:
TASK-56678: avoid displaying backup popup when using Metamask wallet (#233)

avoid displaying Backup popup when using Metamask wallet

ArfaouiMohamedAmine committed:
TASK-55733: Fix Technical label when hovering on wallet address and QRCode from settings page. (#231)

The i18n configuration in the wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/main.js was consuming 'locale/portlet/Portlets_en.properties' instead of 'locale/addon/Wallet_en.properties' until fix has been made, and the button copy in the wallet address was showing a static message 'Copy Address' until a technical label named 'exoplatform.wallet.label.copyAddress' has been added to the 'locale/addon/Wallet_en.properties' .

ahamdi commented on wallet-webapps-common/src/main/webapp/vue-app/components/SendTokensForm.vue:
```diff
@@ -204,8 +205,18 @@ export default {
+    },
     disabled() {
-      return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken || (!this.storedPassword && (!this.walletPassword || !this.walletPassword.trim().length));
+      if (this.isInternalWallet) {
+        return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken || (!this.storedPassword && (!this.walletPassword || !this.walletPassword.trim().length));
+      } else {
+        return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken;
```
Avoid the if/else block and include the check inside the return expression.

ahamdi commented on wallet-webapps-common/src/main/webapp/vue-app/components/SendTokensForm.vue:
```diff
@@ -403,83 +416,149 @@ export default {
-          }
-          if (savedTransaction) {
+
+      if (this.provider === 'INTERNAL_WALLET') {
```
could you refactor a bit the code to not duplicate instructions ?

GouadriaHanen committed:
TASK-55462: fix transaction history list (#232)

when retrieving the transactions list it filters it with the nonce
so when filtring the transactions with the same nonce from different addresses it will take just one (filtred in HashMap) fixed by adding a new list for the result
also fixed opening transaction drawer from transaction notification

ahamdi reviewed (changes requested):
I commented the PR

MayTekayaa committed:
Task-54252: add feedbacks

ahamdi commented on wallet-webapps-common/src/main/webapp/vue-app/components/SendTokensForm.vue:
```diff
@@ -204,8 +205,14 @@ export default {
+    },
     disabled() {
-      return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken || (!this.storedPassword && (!this.walletPassword || !this.walletPassword.trim().length));
+      return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken || this.isInternalWallet ? (!this.storedPassword && (!this.walletPassword || !this.walletPassword.trim().length)) : false;
```
```suggestion
      return !this.walletAddress || this.loading || !this.gasPrice || !this.recipient || !this.amount || !this.canSendToken || (this.isInternalWallet && (!this.storedPassword && (!this.walletPassword || !this.walletPassword.trim().length)));
```

MayTekayaa committed:
Task-54252: update disabled variable's value
